src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerBase.cs

```diff
@@ -267,6 +288,86 @@ public override Task RemoveFromGroupAsync(string connectionId, string groupName,
             return WriteAckableMessageAsync(message, cancellationToken);
         }
 
+#if NET7_0_OR_GREATER
+        public override async Task<T> InvokeConnectionAsync<T>(string connectionId, string methodName, object[] args, CancellationToken cancellationToken = default)
+        {
+            if (IsInvalidArgument(connectionId))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(connectionId));
+            }
+
+            if (IsInvalidArgument(methodName))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(methodName));
+            }
+
+            var invocationId = ClientInvocationManager.Caller.GenerateInvocationId(connectionId);
+
+            try
+            {
+                var message = AppendMessageTracingId(new ClientInvocationMessage(invocationId, connectionId, _callerId, SerializeAllProtocols(methodName, args, invocationId)));
+                await WriteAsync(message);
+                if (ClientConnectionManager.ClientConnections.TryGetValue(connectionId, out var clientConnectionContext))
+                {
+                    var instanceId = clientConnectionContext.InstanceId;
+                    var task = ClientInvocationManager.Caller.AddInvocation<T>(connectionId, invocationId, instanceId, cancellationToken);
+                    return await task;
+                }
+                else
+                {
+                    throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(connectionId));
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        // Only route server will reach here
+        public override async Task SetConnectionResultAsync(string connectionId, CompletionMessage result)
+        {
+            if (IsInvalidArgument(connectionId))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(connectionId));
+            }
+            if (ClientConnectionManager.ClientConnections.TryGetValue(connectionId, out var clientConnectionContext))
+            {
+                // Determine which manager (Caller / Router) the `result` belongs to.
+                // `TryCompletionResult` returns false when the corresponding invocation is not existing.
+                IClientResultsManager clientResultsManager = null;
+                if (ClientInvocationManager.Caller.TryCompleteResult(connectionId, result))
+                {
+                    clientResultsManager = ClientInvocationManager.Caller;
+                }
+                if (ClientInvocationManager.Router.TryCompleteResult(connectionId, result))
+                {
+                    clientResultsManager = ClientInvocationManager.Router;
+                }
+
+                // Block unknown `results` which belongs to neither Caller nor Router
+                if (clientResultsManager != null)
```

Comment:
L338-349 can be simplified to:
```cs
if (!ClientInvocationManager.Caller.TryCompleteResult(connectionId, result) || !ClientInvocationManager.Router.TryCompleteResult(connectionId, result) { return; }
```

Reply:
The simplification should use `&&` rather than `||` ?
```csharp
if (!ClientInvocationManager.Caller.TryCompleteResult(connectionId, result) 
&& 
!ClientInvocationManager.Router.TryCompleteResult(connectionId, result) { return; }
``` 

do `return ;` when the invocation belongs to neither caller nor router. So `&&`

Reply:
It's better not to simplify here. [Explaination](https://github.com/Azure/azure-signalr/pull/1687#discussion_r1001610069)